### PR TITLE
Small improvements to make setup process easier for OS contributors 

### PR DESCRIPTION
**Pull request by AndrewDryga**
For reasoning on each change please see the commit messages.

### CODE REVIEW

**Comment by AndrewDryga** [comment]
Should I remove or update `scripts/uninstall.sh`? Right now it only works for installation from package managers and the most common way to install is Docker, so maybe we don't want to have scripts nobody is using to not make people confused (like I was).

**Comment by AndrewDryga** [comment]
Does anyone have a hint where this is coming from on each `mix test` and `mix.phx server` run?
```
$ mix phx.server
usage: route [-dnqtv] command [[modifiers] args] # <-------
[debug] .....
```

UPD: Found it: https://github.com/firezone/firezone/blob/master/config/dev.exs#L34.

**Comment by jamilbk** [comment]
Wow — thanks for the contribution! Traveling today but should have some time to review later.

> Should I remove or update `scripts/uninstall.sh`? Right now it only works for installation from package managers and the most common way to install is Docker, so maybe we don't want to have scripts nobody is using to not make people confused (like I was).

It may be better to rename the script `scripts/omnibus_uninstall.sh` (and update docs at https://docs.firezone.dev/administer/uninstall/ ) as we still have a large number of Omnibus instances in the wild.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -139,7 +137,7 @@ To start the local development cluster, follow these steps:
 ```
 docker compose build
 docker compose up -d postgres
-docker compose run --rm elixir mix ecto.setup
+docker compose run --rm firezone mix ecto.setup
```
Comment (jamilbk) [inline CONTRIBUTING.md:140]: Thanks for the fix!

File: config/dev.exs [hunk 2]
```diff
@@ -31,8 +31,15 @@ config :fz_http, FzHttpWeb.Endpoint,
   ]
 
 get_egress_interface = fn ->
-  egress_interface_cmd = "route | grep '^default' | grep -o '[^ ]*$'"
-  System.cmd("/bin/sh", ["-c", egress_interface_cmd]) |> elem(0) |> String.trim()
+  egress_interface_cmd =
```
Comment (jamilbk) [inline config/dev.exs:34]: Thanks for improving this!

File: .github/workflows/omnibus_build.yml [hunk 3]
```diff
@@ -100,7 +100,7 @@ jobs:
       - name: Cleanup
         if: always()
         run: |
-          sudo scripts/uninstall.sh
+          sudo scripts/omnibus/omnibus-uninstall.sh
```
Comment (jamilbk) [inline .github/workflows/omnibus_build.yml]: I think you may have an extra `/omnibus` in the path here.

File: apps/fz_http/lib/fz_http/devices.ex [hunk 4]
```diff
@@ -63,24 +63,17 @@ defmodule FzHttp.Devices do
...
-        %Device{}
-        |> Device.create_changeset(attrs)
-        |> Repo.insert()
-      end)
-
-    case result do
-      {:ok, _device} ->
+    attrs
+    |> Device.create_changeset()
+    |> Repo.insert()
```
Comment (jamilbk) [inline apps/fz_http/lib/fz_http/devices.ex:68]: It's not pretty -- but this was in a transaction to prevent a race condition where two devices attempt to be created with the same IP. We [fetch the next available](https://github.com/firezone/firezone/blob/master/apps/fz_http/lib/fz_http/devices/device.ex#L67) IP in the `create_changeset/1`. It's possible two `create_changeset/1` could be called "simultaneously" and end up with the same IP, causing one to fail in the `Repo.insert/1`.

We do have a unique constraint on the column, so maybe we just let the create fail in this edge case. The device tests are run with `async: false` to avoid this race condition, which happens occasionally on fast systems with many cores.

The solution I had wanted to implement was an auto-incrementing SERIAL index for this (which works for INET types in Postgres), but since the IP ranges are configurable by the user, the SERIAL range would need to change at runtime... some sort of runtime migration would be necessary (eek).

I think we've solved this with client apps by using a large, static IP pool (such as CGNAT, `100.64.0.0/10` -- > 4 million addresses) and not letting the pool be configurable as it is now. Client apps would allow Firezone to renumber all client IP addresses at arbitrary times, which isn't possible today since it would require re-importing the static WireGuard config.

**Review by jamilbk — COMMENTED** [review]
LGTM, just a couple notes

File: apps/fz_http/lib/fz_http/devices.ex (hunk 4 above)
Comment (AndrewDryga) [reply apps/fz_http/lib/fz_http/devices.ex:68]: Oh, I did not see the IPvX queries. But..

I don't think wrapping the query in a `Repo.transaction/1` did actually change anything. Postgres uses a [read-committed](https://www.postgresql.org/docs/current/transaction-iso.html#XACT-READ-COMMITTED) isolation level, which means that two separate database transactions can execute two queries and have the same result, and then still have a conflict during the `Repo.insert/1`. I also noticed the comments there that next IP generation should be an insert, which also won't solve the problem.

I like the SERIAL solution, to update it you don't need a migration, it's as simple as `SELECT setval('device_ipv4_seq', (SELECT max(ipv4) FROM devices), true);` when the configuration is changed.

Another way would be to match on function return value and if it's a changeset with a conflict on ipv4/6 fields we can simply retry the function execution, it would be less optimal than a serial though but will be very simple to implement.

Comment (AndrewDryga) [reply apps/fz_http/lib/fz_http/devices.ex:68]: I suggest that we remove the transaction wrap now (as it should not do anything anyways) and create an issue to fix the underlying issue there and to make tests async, I can take that a bit later.

Comment (jamilbk) [reply apps/fz_http/lib/fz_http/devices.ex:68]: 👍

**Review by jamilbk — APPROVED** [review]
LGTM

File: apps/fz_http/lib/fz_http/devices.ex (hunk 4 above)
Comment (jamilbk) [reply apps/fz_http/lib/fz_http/devices.ex:68]: Ah great, didn't realize `SERIAL` was that flexible. That's awesome.

**Comment by AndrewDryga** [comment]
❤️
